.travis.yml

```diff
@@ -1,53 +1,96 @@
 language: python
 
-python:
-  - "3.5"
-  - "3.6"
+sudo: true
+
+python: "3.6"
+
+addons:
+  apt:
+    packages:
+      - build-essential
+      - cmake
+      - libltdl7-dev
+      - libgsl0-dev
+      - libncurses5-dev
+      - libreadline6-dev
+      - pkg-config
+      - python-all-dev
+      - python-matplotlib
+      - python-numpy
+      - python-scipy
 
 before_install:
+  # in before_install, we install all the NESTML dependencies
+
+  # install antrl4
   - sudo apt install openjdk-8-jre
   - wget http://www.antlr.org/download/antlr-4.7.1-complete.jar
   - echo \#\!/bin/bash > antlr4
   - echo java -cp \"`pwd`/antlr-4.7.1-complete.jar:$CLASSPATH\" org.antlr.v4.Tool \"\$@\" >> antlr4
   - echo >> antlr4
   - chmod +x antlr4
   - export PATH=$PATH:`pwd`
-  - pip install --upgrade pip enum34 pytest
+
+  # install Python module dependencies
+  - pip install --upgrade pip pytest
   - pip install -r requirements.txt
- 
-  # gsl library to support ode-toolbox stiffness testing
-  - wget http://ftp.wrz.de/pub/gnu/gsl/gsl-2.5.tar.gz
-  - tar -xvzf gsl-2.5.tar.gz
-  - cd gsl-2.5
-  - ./configure && make && sudo make install
-  - cd ..
-  - export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib
-  - pip install pygsl
-
-  #- pip install matplotlib
-  #- cd ~ && git clone https://github.com/nest/nest-simulator nest_source
-  #- mkdir nest_install
-  #- cd nest_source && mkdir build && cd build
-  #- cmake -DCMAKE_INSTALL_PREFIX=~/nest_install .. && make && make install
-  #- cd $TRAVIS_BUILD_DIR
+
+  # install latest odetoolbox from git master (N.B. GSL/PyGSL installation is skipped as NESTML ignores stiffness test result)
+  - pip uninstall --yes odetoolbox
   - pip install git+https://github.com/nest/ode-toolbox.git
 
 install:
+  # in install, we install NESTML itself
+  - export PYTHONPATH=$PYTHONPATH:`pwd`
+  - echo $PYTHONPATH
+  - python setup.py install
+
+before_script:
+  # use antlr4 to re-generate lexer+parser
+  - cd $TRAVIS_BUILD_DIR
   - find pynestml/generated -not -name __init__.py -a -not -name generated -delete
   - cd pynestml/grammars
   - ./generate_lexer_parser
   - cd ../..
-  - export PYTHONPATH=$PYTHONPATH:`pwd`
-  - echo $PYTHONPATH
-  - python setup.py install
 
-  #- python PyNestML.py -path models -target target
-  #- cd ../target && mkdir build && cd build
-  #- cmake -Dwith-nest=~/nest_install/bin/nest-config .. && make && make install
-  #- cd $TRAVIS_BUILD_DIR
-  #- source $TRAVIS_BUILD_DIR/nest_install/bin/nest_vars.sh
+stages:
+  - test
+  - integration
 
-sudo: true
+jobs:
+  include:
+    - stage: test
+      script: pytest -s tests
+
+    - stage: integration
+      install:
+        # install nest-simulator
+        - pip install cython
+        - cd
+        - git clone --depth=1 https://github.com/nest/nest-simulator
+        - mkdir nest_install
+        - cd nest_install
+        - export PYTHON_INCLUDE_DIR=`python3 -c "from sysconfig import get_paths; info = get_paths(); print(info['include'])"`
```

Comment:
I'm surprised, was that really necessary? `-DPYTHON_LIBRARY` should not be required on a typical ubuntu machine...

Reply:
Unfortunately, the Travis instances are set up to paths like `/home/travis/virtualenv/python3.6.7/include/python3.6m`. These are not searched by default. I would prefer not to hard-code the exact path or version number. This is the one-liner solution I came up with.